docs/upgrade-notes/update-cumulus_id-type-indexes-CUMULUS-3449.md

```diff
@@ -0,0 +1,176 @@
+# Last window
+<Ctrl>-b l
+
+# Next Window
+<Ctrl>-b n
+ 
+# Detach from tmux session
+<Ctrl>-b d
+ 
+# List tmux sessions
+tmux ls
+
+# Reattach to the session, you will see the windows again
+tmux attach -t CumulusUpgrade
+
+# Exit the window and/or close/kill the session
+# Warning: Don't close/kill the session until the task is complete
+exit
+# or
+<Ctrl>-b x
+```
+
+## Upgrade Steps
+
+1. Quiesce ingest
+
+    Stop all ingest operations in Cumulus Core according to your operational procedures. You should validate
+    that it appears there are no active queries that appear to be inserting granules/files into the database
+    as a secondary method of evaluating the database system state:
+
+    ```text
+    select pid, query, state, wait_event_type, wait_event from pg_stat_activity where state = 'active';
+    ```
+
+    If query rows are returned with a `query` value that involves the tables, make sure ingest is halted
+    and no other granule-update activity is running on the system.
+
+    Note: In rare instances if there are hung queries that are unable to resolve, it may be necessary to
+    manually use psql [Server Signaling
+    Functions](https://www.postgresql.org/docs/13/functions-admin.html#FUNCTIONS-ADMIN-SIGNAL)
+    `pg_cancel_backend` and/or
+    `pg_terminate_backend` to end the queries.
+
+2. Login into EC2 instance with database access.
+
+    From AWS console, go to EC2, pick a `<prefix>-CumulusECSCluster` instance, click Connect, click Session Manager.
+    :::note Remember to take a note on which instance you run the commands. :::
+
+3. Install tmux and postgres client
+
+    ```sh
+    sudo yum install -y tmux postgresql.x86_64
+    ```
+
+    Once installed, a tmux session is started with two windows. The Cumulus database is connected to in each window
+    using the PostgreSQL client. The primary window is used for running the ALTER commands, while the secondary window
+    is used to monitor the database and alter statement. When the operator hits end of shift or is done monitoring for
+    the day, the tmux session can be detached from and reattached to at a later time.
+
+4. Run SQL Commands
+    The database login credentials can be retrieved from the prefix_db_login secret.
+
+    ```sh
+    tmux new-session -s CumulusUpgrade -n Fix-DataTypes
+
+    psql -h <Endpoint for writer instance> -p <Port for database or 5432> -d <cumulus database name> -U <database admin user> -W
+    #e.g. psql -h cumulus-dev-rds-cluster.cluster-xxx.us-east-1.rds.amazonaws.com -p 5432 -d cumulus_test_db -U cumulus_test -W
+
+    # Update column types
+    ALTER TABLE executions ALTER COLUMN cumulus_id TYPE BIGINT, ALTER COLUMN parent_cumulus_id TYPE BIGINT;
+    ALTER TABLE files ALTER COLUMN granule_cumulus_id TYPE BIGINT;
+    ALTER TABLE granules_executions ALTER COLUMN granule_cumulus_id TYPE BIGINT, ALTER COLUMN execution_cumulus_id TYPE BIGINT;
+    ALTER TABLE pdrs ALTER COLUMN execution_cumulus_id TYPE BIGINT;
```

Comment:
So pdrs cumulus_id will never be big enough or run into the type mismatch? (I don't know much about pdrs so just wondering)

Reply:
Most DAACs don't use PDRs, and one PDR can have multiple granules.